### PR TITLE
Fix self delegation and added errors in form

### PR DESCRIPTION
**Pull request by mllocs**
Adds form errors when creating a Delegation. Also prevents admins delegating votes to themselves. Fixes https://github.com/coopdevs/decidim-module-action_delegator/issues/12.

### CODE REVIEW

File: app/commands/decidim/action_delegator/admin/create_delegation.rb [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+        private
+
+        attr_reader :form, :performed_by, :current_setting
+
+        def self_delegate?
+          return false unless performed_by.id == form.grantee_id
+
+          form.errors.add(:grantee_id, :self_delegate)
+          true
+        end
```
Comment (sauloperez) [inline app/commands/decidim/action_delegator/admin/create_delegation.rb:40]: By looking at how Decidim does it, this validation method should be a responsibility of the form object, right? then, this command will be concerned only about creating the delegation.

**Review by sauloperez — CHANGES_REQUESTED** [review]

File: app/commands/decidim/action_delegator/admin/create_delegation.rb (hunk 1 above)
Comment (microstudi) [reply app/commands/decidim/action_delegator/admin/create_delegation.rb:40]: yes, that's why the command broadcasts `:invalid` if the form is invalid

File: app/forms/decidim/action_delegator/admin/delegation_form.rb [hunk 2]
```diff
@@ -0,0 +1,25 @@
+# frozen_string_literal: true
+
+module Decidim
+  module ActionDelegator
+    module Admin
+      # A form object used to create a Delegation
+      #
+      class DelegationForm < Form
```
Comment (microstudi) [inline app/forms/decidim/action_delegator/admin/delegation_form.rb:8]: We should a rspec for this class

File: app/commands/decidim/action_delegator/admin/create_delegation.rb [hunk 3]
```diff
@@ -0,0 +1,49 @@
+# frozen_string_literal: true
+
+module Decidim
+  module ActionDelegator
+    module Admin
+      class CreateDelegation < Rectify::Command
```
Comment (microstudi) [inline app/commands/decidim/action_delegator/admin/create_delegation.rb:6]: Needs a spec this class

File: app/commands/decidim/action_delegator/admin/create_delegation.rb (hunk 1 above)
Comment (mllocs) [reply app/commands/decidim/action_delegator/admin/create_delegation.rb:40]: @sauloperez I think the form object only validates the attributes, without much other context. I think form objects as a mixture of strong params and model validations. 

Validating that admins don't grant to themselves feels like it belongs to another place. I also was heavily inspired by this other command https://github.com/decidim/decidim/blob/develop/decidim-meetings/app/commands/decidim/meetings/admin/invite_user_to_join_meeting.rb (the `already_invited?` validation looked very similar to our `self_delegate?`).

File: app/forms/decidim/action_delegator/admin/delegation_form.rb (hunk 2 above)
Comment (mllocs) [reply app/forms/decidim/action_delegator/admin/delegation_form.rb:8]: done

File: app/commands/decidim/action_delegator/admin/create_delegation.rb (hunk 3 above)
Comment (mllocs) [reply app/commands/decidim/action_delegator/admin/create_delegation.rb:6]: done

File: app/commands/decidim/action_delegator/admin/create_delegation.rb [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+        # Public: Initializes the command.
+        #
+        # form         - A form object with the params.
+        # delegated_by - The user performing the operation
```
Comment (sauloperez) [inline app/commands/decidim/action_delegator/admin/create_delegation.rb:10]: :warning: this argument name doesn't match the actual method argument

**Review by sauloperez — APPROVED** [review]

**Comment by sauloperez** [comment]
Looks better now :ok_hand:

**Review by sauloperez — APPROVED** [review]

**Pull request merged**